sjb/children.sh

```diff
@@ -4,21 +4,24 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-function children() {
-	local config=$1
-
-	for child_config in sjb/config/test_cases/*.yml sjb/config/common/test_cases/*.yml; do
-		if ! grep -q 'parent' "${child_config}"; then
-			continue
-		fi
+CONFIGS='sjb/config/test_cases/*.yml sjb/config/common/test_cases/*.yml'
+declare -A PARENTS
 
-		local parent_config
-		parent_config="sjb/config/$( grep -Po "(?<=parent: ')[^']+(?=')" "${child_config}" )"
-		if [[ "${parent_config}" == "${config}" ]]; then
-			echo "${child_config}"
-			children "${child_config}"
-		fi
-	done
+function children() {
+    local config=$1 child_config
+    for child_config in $CONFIGS; do
+        if [[ "${PARENTS[$child_config]:-}" == "$config" ]]; then
+            echo "${child_config}"
+            children "${child_config}"
+        fi
+    done
 }
 
-children "$1"
\ No newline at end of file
+for config in $CONFIGS; do
+    if parent=$(grep -Po "(?<=parent: ')[^']+(?=')" "${config}"); then
```

Comment:
We could use Python and `yaml.load()` :)

Reply:
See below.